### PR TITLE
Feature: AWS CloudWatch Container Insights Addon

### PR DESCRIPTION
**Pull request by 5herlocked**
*Issue #, if available:* #872

*Description of changes:*
Added container insights add-on using CoreAddOn.
Added documentation for container insights along with what it's doing.

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: lib/addons/cloud-watch-insights/index.ts [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+    deploy(clusterInfo: ClusterInfo): Promise<Construct> {
+      const cluster = clusterInfo.cluster;
+      const context = clusterInfo.getResourceContext();
+
+      const insightsSA = cluster.addServiceAccount("CloudWatchInsightsSA", {
+        name: `containers-insights-sa`,
+        namespace: defaultProps.namespace
+      });
+
+      const insightsAddon = new eks.CfnAddon(context.scope,  "CloudWatchInsightsAddon", {
```
Comment (elamaran11) [inline lib/addons/cloud-watch-insights/index.ts:56]: You should be adding CloudWatch Managed Policy like [this](https://github.com/aws-quickstart/cdk-eks-blueprints/blob/6ec2a2403ef34052c7d898f6d72a436b50fd589b/lib/addons/container-insights/index.ts#L35).

File: lib/addons/cloud-watch-insights/index.ts [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+export type CloudWatchInsightsAddOnProps = Omit<CoreAddOnProps, "saName" | "addOnName" | "version"> & {
+  /**
+   * Gives CloudWatch agent access to EBS performance systems by adding an IAM role as defined here:
+   * https://docs.aws.amazon.com/AmazonCloudWatch/latest/monitoring/install-CloudWatch-Observability-EKS-addon.html#install-CloudWatch-Observability-EKS-addon-configuration
+   */
+  ebsPerformanceLogs?: boolean,
+  /**
+   * Custom CloudWatch Agent configuration, specifics can be found here:
+   * https://docs.aws.amazon.com/AmazonCloudWatch/latest/monitoring/install-CloudWatch-Observability-EKS-addon.html#install-CloudWatch-Observability-EKS-addon-configuration
+   */
```
Comment (elamaran11) [inline lib/addons/cloud-watch-insights/index.ts:21]: [Container Logs](https://docs.aws.amazon.com/AmazonCloudWatch/latest/monitoring/install-CloudWatch-Observability-EKS-addon.html#install-CloudWatch-Observability-EKS-addon-configuration:~:text=Optional)%20Additional%20configuration-,Container%20logs,-By%20default%2C%20the) option is missing. This means we can have a conflicts with CW Logs addon?

File: docs/addons/aws-container-insights.md [hunk 3]
```diff
@@ -0,0 +1,111 @@
...
+Add the following as an add-on to your main.ts file to add CloudWatch Insights to your cluster
+
+```typescript
+import 'source-map-support/register';
+import * as cdk from 'aws-cdk-lib';
+import * as blueprints from '@aws-quickstart/eks-blueprints';
+
+const app = new cdk.App();
+
+const addOn = new blueprints.addons.CloudWatchInsights();
```
Comment (elamaran11) [inline docs/addons/aws-container-insights.md:26]: Could we show additional different variants of usage like EBS, CW Logs etc?

File: lib/addons/cloud-watch-insights/index.ts [hunk 4]
```diff
@@ -0,0 +1,75 @@
...
+
+  readonly options: CloudWatchInsightsAddOnProps;
+
+    constructor(props?: CloudWatchInsightsAddOnProps) {
+      super({ ...defaultProps, ...props });
+
+      this.options = props ?? {};
+    }
+
+    @conflictsWith("AdotCollectorAddon", "CloudWatchAdotAddon")
```
Comment (elamaran11) [inline lib/addons/cloud-watch-insights/index.ts:46]: `CloudWatchLogsAddon` is missing with `conflictsWith`

File: docs/addons/aws-container-insights.md [hunk 5]
```diff
@@ -0,0 +1,111 @@
...
+amazon-cloudwatch   fluent-bit                                           1/1     Running   1 (4d22h ago)   5d
+```
+
+## View metrics for cluster and workloads
+
+Under Performance Monitoring, the Container Insights dashboard allows you to hone in on both cluster and workload metrics. After selecting EKS Pods and Clusters, you will see that the dashboard provides CPU and memory utilization along with other important metrics such as network performance.
+
+![CloudWatch](./../assets/images/eks-blueprint-cwinsights-performance-monitoring.png)
+
+## View cluster level logs
```
Comment (elamaran11) [inline docs/addons/aws-container-insights.md:57]: Could we use doc and screenshots aligning to [the](https://aws.amazon.com/blogs/mt/new-container-insights-with-enhanced-observability-for-amazon-eks/) blog :
- Container Insights – Dashboard
- Top 10 list view
- Clusters overview
- Clusters performance monitoring
- CPU and memory utilization at the node level
- Containers performance monitoring
- Nodes performance monitoring
- Services performance monitoring
- Dashboard views for popular workloads
- New “ContainerInsights” metrics namespace
- Enhanced Container Insights – various metrics
- CW Logs
- EBS Performance
- Alarms?

**Review by elamaran11 — CHANGES_REQUESTED** [review]
@5herlocked Nice work on this new Addon. Some feedback.
1. Doc can be improved aligning to enhanced CW features of the addon
2. Managed Policy is missing
3. CW logs functionality is missing
4. mkdocs is missing
5. adding to doc index is missing
6. Are we not adding the blueprints stack. I would add and comment it.

**Comment by elamaran11** [comment]
@5herlocked Wanted to check if you need any help to make progress on this addon. We want to add this with next blueprints release.

**Comment by 5herlocked** [comment]
I'll have this completed in a few weeks.

**Comment by elamaran11** [comment]
> I'll have this completed in a few weeks.

Could you plan to add this for upcoming blueprints release?

**Comment by 5herlocked** [comment]
Resolved most of the requested changes, still working on the documentation.

**Review by elamaran11 — APPROVED** [review]
LGTM. @shapirov103 Please check from your end.

**Review by shapirov103 — COMMENTED** [review]
@5herlocked 

Looks great, however the way it stands now it will be very confusing for the customers to select the right addon, especially choosing between ContainerInsightsAddon and this addon. 

So let's apply the following changes:

1. deprecate the current container insights addon (e..g similar to what we did for calico)
2. update documentation for the deprecated addon and specify that there will be no maintenace on it and customers should move over to the CloudWatchInsightsAddon. 
3. In the docs for this addon, also call out that it is replacing the current CI addon.

**Review by shapirov103 — CHANGES_REQUESTED** [review]
Please see the recommended naming and deprecation changes, otherwise looks great.

**Review by elamaran11 — COMMENTED** [review]
Nice work, Please make sure the screenshots are up-to-date after your testing.

**Comment by elamaran11** [comment]
@shapirov103 I was able to pull the fork and test, everything works fine. I can see CI in CW Console. Please proceed with e2e and merge.

**Pull request merged**